src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveMetadataService.cs

```diff
@@ -95,7 +95,7 @@ public async Task<RetrieveMetadataResponse> RetrieveSopInstanceMetadataAsync(str
     private RetrieveMetadataResponse RetrieveMetadata(IReadOnlyList<VersionedInstanceIdentifier> instancesToRetrieve, bool isCacheValid, string eTag, CancellationToken cancellationToken)
     {
         _contextAccessor.RequestContext.PartCount = instancesToRetrieve.Count;
-        _telemetryClient.TrackInstanceCount(instancesToRetrieve.Count);
+        _instanceMeter.InstanceCount.Add(instancesToRetrieve.Count);
```

Comment:
Seeing this only on retrieval makes me think we may need to rename some of the metric names to make their usage more apparent

Reply:
I added a new instrument to track the instance count in Retrieve function.